### ISSUE TITLE
Applying subschemas conditionally in combination with additionalProperties:false

### ISSUE THREAD

**Issue by corleymanz** [JamesNK/Newtonsoft.Json.Schema#203]
I want to use conditionally subschema's in combination with additionalProperties set to false. Is this possible or is this a bug? See below schema and json. I used youre website to validate this https://www.jsonschemavalidator.net/

json schema:
{
  "type": "object",
    "additionalProperties" : false,
  "properties": {
    "street_address": {
      "type": "string"
    },
    "country": {
      "enum": ["United States of America", "Canada"]
    }
  },
  "if": {
    "properties": { "country": { "const": "United States of America" } }
  },
  "then": {
    "properties": { "postal_code": { "type": "string" } }
  },
  "else": {
    "properties": { "postal_code": { "type": "string" } }
  }
}

json test example
{
  "street_address": "1600 Pennsylvania Avenue NW",
  "country": "United States of America",
  "postal_code": "1"
}

errors found (on https://www.jsonschemavalidator.net/):
Property 'postal_code' has not been defined and the schema does not allow additional properties.

**Comment by corleymanz** [comment]
I just read the specs and i think we need to wait for tghe next version

https://json-schema.org/understanding-json-schema/reference/combining.html

```
Unfortunately, now the schema will reject everything. This is because the Properties refers to the entire schema. And that entire schema includes no properties, and knows nothing about the properties in the subschemas inside of the allOf array.

This shortcoming is perhaps one of the biggest surprises of the combining operations in JSON schema: it does not behave like inheritance in an object-oriented language. There are some proposals to address this in the next version of the JSON schema specification.
```

**Comment by handrews** [comment]
@corleymanz yes, that is correct.  `unevaluatedProperties` in draft 2019-09 was created to address this use case, which has been a longstanding (like since at least 2012ish) complaint.

Draft 2019-09 ([formerly known as draft-08](https://json-schema.org/specification-links.html#table-of-all-versions-of-everything)) has quite a lot in it, and we are still working on finishing the test suite (help welcome if anyone wants to volunteer!).

2019-09 (or possibly a very minor update 2020-03 that just clarifies 2019-09 based on early feedback) will be adopted by OpenAPI 3.1, so we expect it to see a lot of use.  It should be a stable base for the final drive to standardization, as it (finally) includes an extensibility mechanism which has been a big shortcoming.  Without extensibility, everyone wants their own favorite keyword in the spec which means we'd never finish.  That was the main reason for the long wait between draft-07 and 2019-09.
